src/modal/testing/modal.service.spec.ts

```diff
@@ -37,47 +37,69 @@ describe('Modal service', () => {
     fixture.detectChanges();
   });
 
-  it('should return random id on spin up a new modal', done => {
+  it('should return random id on spin up a new modal', () => {
     modalService.onShown.subscribe((data) => {
       expect(data.id).toBeTruthy();
-      done();
```

Comment:
In my experience, your tests must call the `done` callback when you test async operations.

The problem is that a test without an `expect` will pass as successfully. For instance, this counts as a successful test:

```ts
it('should pass', () => {
  const foo = 'bar';
})
```

The test runs, and if it doesn't find any `expect` at the end, it will pass as valid.
This example will also pass, even though we want it to fail.

```ts
it('should fail', () => {
  setTimeout(() => {
    expect(true).toBe(false);
  }, 3000);
});
```

It will run, put an async operation on queue, finds the end of the test, no `expect` found, test successful, and our test never takes our assert into account.

---

To tell the test that the assert executed after an async operation, we must use the `done` callback

```ts
it('should fail', (done) => {
  setTimeout(() => {
    expect(true).toBe(false);
    done();
  }, 3000);
});
```

Reply:
Thanks for clarification! Bringing them back!

Reply:
Thank you for the detailed message!